packages/router/history.ts

```diff
@@ -536,11 +540,17 @@ export function createLocation(
   state: any = null,
   key?: string
 ): Readonly<Location> {
+  let toLocation = typeof to === "string" ? parsePath(to) : to;
+  if (toLocation.pathname) {
+    toLocation.pathname = normalizeMultipleSlashes(toLocation.pathname);
+  }
   let location: Readonly<Location> = {
-    pathname: typeof current === "string" ? current : current.pathname,
+    pathname: normalizeMultipleSlashes(
+      typeof current === "string" ? current : current.pathname
+    ),
     search: "",
     hash: "",
-    ...(typeof to === "string" ? parsePath(to) : to),
+    ...toLocation,
```

Comment:
I'm curious if we think this is too widespread.  There is a very targeted fix we could put in `createURL` that just fixes scenarios we expect would throw.  By doing it here are a bit more robust and effectively sanitize when we create `Location`'s for the router.  We do then expose `Location`'s back out to the user so it might be a concern of that feeling "breaking" to mess with incoming paths in a different manner...